api/src/main/java/org/openmrs/module/reporting/report/renderer/JSONTemplateRenderer.java

```diff
@@ -0,0 +1,144 @@
+import org.openmrs.module.reporting.report.ReportRequest;
+import org.openmrs.module.reporting.report.renderer.template.TemplateEngine;
+import org.openmrs.module.reporting.report.renderer.template.TemplateEngineManager;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Report Renderer implementation that supports rendering of a text template
+ */
+@Handler
+@Localized("reporting.JSONTemplateRenderer")
+public class JSONTemplateRenderer extends ReportTemplateRenderer {
+	
+	public static final String TEMPLATE_TYPE = "templateType";
+	
+	private Log log = LogFactory.getLog(this.getClass());
+	
+	public JSONTemplateRenderer() {
+		super();
+	}
+
+	/**
+	 * @see ReportRenderer#getRenderedContentType(org.openmrs.module.reporting.report.ReportRequest)
+	 */
+	public String getRenderedContentType(ReportRequest request) {
+		String contentType = super.getRenderedContentType(request);
+		if (StringUtils.isEmpty(contentType)) {
+			contentType = "application/json";
+		}
+		return contentType;
+	}
+	
+	/**
+	 * @see ReportTemplateRenderer#getBaseReplacementData(ReportData, ReportDesign)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public Map<String, Object> getBaseReplacementData(ReportData reportData, ReportDesign design) {
+		// Populate the replacement data with all core values, and any data sets with only one row
+		Map<String, Object> data = super.getBaseReplacementData(reportData, design);
+		
+		// Now go through and add data sets and add rows by index to replacement data as well
+		for (String dataSetName : reportData.getDataSets().keySet()) {
+			DataSet ds = reportData.getDataSets().get(dataSetName);
+			int rowNum = 0;
+			for (DataSetRow row : ds) {
+				for (Object entry : row.getColumnValues().entrySet()) {
+					rowNum++;
+					Map.Entry<DataSetColumn, Object> e = (Map.Entry<DataSetColumn, Object>) entry;
+					String baseKey = dataSetName + SEPARATOR + e.getKey().getName() + SEPARATOR + rowNum;
+					Object replacementValue = getReplacementValue(e.getValue());
+					data.put(baseKey, replacementValue);
+					String columnLabel = Context.getMessageSourceService().getMessage(e.getKey().getLabel());
+					data.put(baseKey + SEPARATOR + LABEL, columnLabel);
+					if (reportData.getDataSets().size() == 1) {
+						data.put(e.getKey().getName() + SEPARATOR + rowNum, replacementValue);
+						data.put(e.getKey().getName() + SEPARATOR + rowNum + SEPARATOR + LABEL, columnLabel);
+					}
+				}				
+			}
+		}
+		return data;
+	}
+	
+	/**
+	 * @see ReportRenderer#render(ReportData, String, OutputStream)
+	 */
+	public void render(ReportData reportData, String argument, OutputStream out) throws IOException, RenderingException {
+		
+		log.debug("Attempting to render report with JSONTemplateRenderer");
+		
+		Writer pw = new OutputStreamWriter(out, "UTF-8");
```

Comment:
Are you trying to write he objecst to json ?? could you use libraries like 
```
org.codehaus.jackson.JsonParser;
 org.codehaus.jackson.map.ObjectMapper;
```

Reply:
@mozzy11 this template leverages the existing TextTemplateRenderer, we just needed to configure it to render json, which it does

Reply:
alright. Its ok if youve achieved your objective then. you can then fix the failing tests